src/engine/enginepregain.cpp

```diff
@@ -51,36 +53,46 @@ EnginePregain::EnginePregain(QString group)
 EnginePregain::~EnginePregain()
 {
     delete m_pPotmeterPregain;
-    delete m_pControlReplayGain;
+    delete m_pCOReplayGain;
     delete m_pTotalGain;
 
     delete s_pEnableReplayGain;
     s_pEnableReplayGain = NULL;
     delete s_pReplayGainBoost;
     s_pReplayGainBoost = NULL;
+    delete s_pDefaultBoost;
+    s_pDefaultBoost = NULL;
 }
 
 void EnginePregain::process(CSAMPLE* pInOut, const int iBufferSize) {
-
-    float fEnableReplayGain = s_pEnableReplayGain->get();
-    const float fReplayGainBoost = s_pReplayGainBoost->get();
-    float fGain = m_pPotmeterPregain->get();
-    float fReplayGain = m_pControlReplayGain->get();
-    float fReplayGainCorrection = 1;
-    float fPassing = m_pPassthroughEnabled->get();
-
-    // Override replaygain value if passing through
-    if (fPassing > 0) {
-        fReplayGain = 1.0;
-    } else if (fReplayGain * fEnableReplayGain != 0) {
-        // Here is the point, when ReplayGain Analyser takes its action, suggested gain changes from 0 to a nonzero value
+    const float fReplayGain = m_pCOReplayGain->get();
+    float fReplayGainCorrection;
+    if (m_pPassthroughEnabled->toBool()) {
+        // Override replaygain value if passing through
+        // TODO(XXX): consider a good default.
+        // Do we expect an replaygain leveled input or
+        // Normalized to 1 input?
+        fReplayGainCorrection = 1; // We expect a replaygain leveled input
+    } else if (!s_pEnableReplayGain->toBool() || fReplayGain == 0) {
```

Comment:
After seeing the list RJ compiled of other software all using master headroom + limiter, it seems that's the standard model and we should follow that.  I still think the headroom should be smaller, just a few db, but that's an easy preference setting.

Note that we never had a master headroom setting -- we had a per-channel hard-coded headroom by mistake.

Mic and aux have their own pregain controls so there's no need for a boost setting.  The replaygain boost is unique because it compensates for replaygain's tendency to make things soft.

Also RJ pointed out to me that we do not do any clipping at the channel stage -- just in the master mix.  so peak indicators on individual channels are not a problem for sound quality.